mlir/test/Gradient/BufferizationTest.mlir

```diff
@@ -27,3 +27,27 @@ func.func @adjoint(%arg0: f64, %arg1: index) {
     %grad = gradient.adjoint @circuit(%arg0) size(%arg1) : (f64) -> tensor<?xf64>
     return
 }
+
+///
+
+func.func private @circuit2(%arg0: f64)
+
+func.func @backprop(%arg0: f64, %arg1: index) {
+
+    // CHECK:   [[alloc:%.+]] = memref.alloc(%arg1) : memref<?xf64>
+    // CHECK:   gradient.backprop @circuit2(%arg0) size(%arg1) in([[alloc]] : memref<?xf64>) : (f64) -> ()
+    %grad = gradient.backprop @circuit2(%arg0) size(%arg1) : (f64) -> tensor<?xf64>
+    return
+}
+
+///
+
+func.func private @circuit3(%arg0: tensor<?x2xf64>)
+
+func.func @backprop2(%arg0: tensor<?x2xf64>, %arg1: index) {
+
+    // CHECK:   [[alloc:%.+]] = memref.alloc(%dim, %arg1) : memref<?x?xf64>
+    // CHECK:   gradient.backprop @circuit3(%0) size(%arg1) in(%alloc : memref<?x?xf64>) : (memref<?x2xf64>) -> ()
+    %grad = gradient.backprop @circuit3(%arg0) size(%arg1) : (tensor<?x2xf64>) -> tensor<?x?xf64>
+    return
+}
```

Comment:
It might make sense to add another test here so the entire bufferization function is covered, for example when there are multiple arguments.

Reply:
Yes I will add more tests